src/controller/CHIPDeviceController.cpp

```diff
@@ -2955,6 +3016,47 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+    case CommissioningStage::kICDSymmetricKeyGeneration: {
+        if (mICDRegistrationDelegate == nullptr)
+        {
+            ChipLogError(Controller, "No ICD Registration Delegate provided, the commissioner cannot do ICD registration!");
+            CommissioningStageComplete(CHIP_ERROR_INCORRECT_STATE);
+            return;
+        }
+
+        mICDRegistrationDelegate->GenerateSymmetricKey(proxy->GetDeviceId(),
+                                                       &mICDRegistrationOnSymmetricKeyGenerationCompletedCallback);
+    }
+    break;
+    case CommissioningStage::kICDRegistration: {
+        IcdManagement::Commands::RegisterClient::Type request;
+
+        if (!(params.GetICDControllerNodeId().HasValue() && params.GetICDSubjectId().HasValue() &&
+              params.GetICDSymmetricKey().HasValue()))
+        {
+            ChipLogError(Controller, "No ICD Registration Information provided!");
+            CommissioningStageComplete(CHIP_ERROR_INCORRECT_STATE);
+            return;
+        }
+
+        request.checkInNodeID    = params.GetICDControllerNodeId().Value();
+        request.monitoredSubject = params.GetICDSubjectId().Value();
+        request.key              = params.GetICDSymmetricKey().Value();
+        if (params.GetICDVerificationKey().HasValue())
```

Comment:
Why would there ever need to be a verification key here?  Verification key is only needed for non-administrator use cases _when a prior ICD key was set up_, no?  This situation fulfills neither condition.

Reply:
Thanks for clarification.

In the spec rev.49003c1b2, this field is referred as: "SHALL be optional for clients with administrator permissions". I think it means "...clients with administrator MAY present..."

Do we need update the statement in the spec to indicate that clients with administrator permissions SHOULD NOT present a verification key.

Reply:
We can clarify that bit of spec. Adding the `should not` is not a bad idea.
Can you create a spec issue for this?

Reply:
Created as https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/8519